app/src/main/java/com/arctouch/codechallenge/util/BindingAdapters.kt

```diff
@@ -0,0 +1,44 @@
+package com.arctouch.codechallenge.util
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.Observer
+import android.databinding.BindingAdapter
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.util.extension.getParentActivity
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+
+@BindingAdapter("mutableVisibility")
+fun setMutableVisibility(view: View, visibility: MutableLiveData<Int>?) {
+    val parentActivity: AppCompatActivity? = view.getParentActivity()
+    if(parentActivity != null && visibility != null) {
```

Comment:
An alternative to checking if the parameters provided to the method are "acceptable" (in this case they must be non-null) is adding restrictions to the method signature, such as:
```
fun setMutableVisibility(view: View, parentActivity: Activity, visibility: MutableLiveData<Int>) {
    // code
}
```
Which alternative do you think is better, why? (P.S.: I don't know if this signature is possible here, so imagine it is)

Reply:
In the given signature, the null verification would not be necessary, but that function is being called by its annotated value "@BindingAdapter("mutableVisibility")" inside XML code, so the verification can not be made there and the application would not work well when setting null value to these property.